### PR TITLE
Add premium page +  stripe

### PR DESCRIPTION
**Pull request by mezotv**

### CODE REVIEW

**Comment by mezotv** [comment]
Will need some work @Jester0027 for now I only copied over the old code, will work on this soon!

**Review by Jester0027 — CHANGES_REQUESTED** [review]

File: helpers/cache/redis.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"use server";
+import { Redis } from "@upstash/redis"
+import dotenv from "dotenv";
+dotenv.config();
+
+
+const redis = new Redis({
+  url:  process.env.REDIS_URL!,
+  token: process.env.REDIS_TOKEN!,
```
Comment (Jester0027) [inline helpers/cache/redis.ts:9]: Can you add those environment variable to the example file at the root of the project ?
Could you also add a docker compose environment, or a pod with a redis container with podman ?

**Review by Jester0027 — DISMISSED** [review]

File: helpers/cache/redis.ts (hunk 1 above)
Comment (mezotv) [reply helpers/cache/redis.ts:9]: At some point maybe but honsetly upstash is really cheap so might aswell usw that.

File: components/Homepage/Content.tsx [hunk 2]
```diff
@@ -68,8 +68,12 @@ export function HomeContent({
             I Ever, Higher or Lower, and What Would You Do!{" "}
             <span className="hidden">tempus voicus</span>
```
[the remark this answers is not in the document]
Comment (dpaulos6) [reply components/Homepage/Content.tsx:69]: its a thing dominik made to appear in search engines, if you search for that would you bot will appear.

File: app/premium/_components/ServersList.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const servers: any = await getServer();
+      console.log(servers, "servers");
+      setServersData(servers);
+    };
+
+    fetchData();
+  }, []);
```
Comment (Jester0027) [inline app/premium/_components/ServersList.tsx:21]: Server components can't use hooks.
Fetch the data as the component is being rendered.
```suggestion
      const serversData = await getServer();
```

File: app/premium/page.tsx [hunk 4]
```diff
@@ -77,30 +72,20 @@ const pricingData: PricingData = {
...
   const [isMonthly, setIsMonthly] = useState(true);
   const [serversData, setServersData] = useState<DiscordGuild[]>([]);
 
+  const handleChange = () => {
+    setIsMonthly(!isMonthly);
+  };
+
   const fetchData = async () => {
-    const servers = await getServer();
+    const servers: any = await getServer();
```
Comment (Jester0027) [inline app/premium/page.tsx:84]: Why did you type it as `any` here ?

File: package.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "@upstash/redis": "^1.30.0",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.1",
+    "dotenv": "^16.4.5",
```
Comment (Jester0027) [inline package.json:20]: Nestjs already handles .env files, this dependency is not needed

Comment (mezotv) [reply package.json]: Yea fair, I was using it in a file tho even if it didnt do anything please @dpaulos6 when you remove a package remove everything!

**Pull request merged**